File: quickSort.c
#include <stdio.h> 
#include <stdlib.h>

int partion(int* array, int start, int end)
{
    int i= start;
    int j;
    int x = array[end];

    for (j=start; j<end; j++)
    {
        if (array[j] < x)
        {
            int temp = array[i];
            array[i] = array[j];
            array[j] = temp;
            i++;
        }
    }

    array[end] = array[i];
    array[i] = x;
    return i;
}

int quickSort(int* array, int start, int end)
{
    if (start >= end)
        return;

    int div = partion(array, start, end);
    quickSort(array, start, div-1);
    quickSort(array, div+1, end);
}

int main(int argc, char** argv)
{
    int len = argc -1;
    int array[len];
    int i;
    printf("quick sort\nUnsorted array:"); 
    for(i=0; i<len; i++)
    {
        array[i] = atoi(argv[i+1]);
        printf("%d ", array[i]);
    }
    printf("\n");
    
    quickSort(array, 0, len-1);
    printf("Sorted array :"); 
    for(i=0; i<len; i++)
    {
        printf("%d ", array[i]);
    }
    printf("\n");
    return 1;
}

File: quickSort.py
#!/usr/bin/python
#FileName: quickSort.py

import sys

def partition(array, start, end):
    x = array[end]
    i = start
    for j in range(start, end):
        if int(array[j]) < int(x):
            temp = array[j]
            array[j] = array[i]
            array[i] = temp
            i = i + 1
    array[end] = array[i]
    array[i] = x
    return i

def quickSort(array, start, end):
    if start >= end:
        return
    div = partition(array, start, end)
    quickSort(array, start, div-1)
    quickSort(array, div+1, end)

sArray = []
for i in range(1, len(sys.argv)):
    sArray.append(sys.argv[i])
print 'before sort', sArray

quickSort(sArray, 0, len(sArray)-1)
print 'sorted array:', sArray

